src/libcore/iter.rs

```diff
@@ -2644,6 +2644,16 @@ impl<A: Step + Zero + Clone> Iterator for StepBy<A, ops::Range<A>> {
             None
         }
     }
+
+    #[inline]
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        match Step::steps_between(&self.range.start,
+                                  &self.range.end,
+                                  &self.step_by) {
+            Some(hint) => (hint, Some(hint)),
```

Comment:
Easy fix: `Some(hint) => (hint, Some(hint+1)),`

Can be precisely calculated only if `A` implements `One` and `Sub<Output=A>`
Possible solution (caveat: only works if `A::one()` is less than `step_by`):
```rust
        Some(hint) => {
            let steps = hint + 1;
            let end = &self.range.end - &A::one(); // to check whether range.end was included
            if Step::steps_between(&self.range.start, &end, &step_by) < hint {
                // range.end was included (but range is end-exclusive)
                steps -= 1;
            }
            (steps, Some(steps))
        },
```
An easier solution would be to include `steps_exclusively_between` in `Step` trait. Then we can just add one to it (if there is a result) and return.